doc/voq/fabric.md

```diff
@@ -171,6 +172,85 @@ PORT            RxCells     TxCells      Crc       Fec  Corrected
 
 In a later phase, a `show fabric status` command will be added to show the remote switch ID and link ID for each fabric link of an ASIC. This will be obtained from the SAI_PORT_ATTR_FABRIC_REACHABILITY port attribute of the fabric port. Note that for fabric links that do not have a link partner because of the configuration of the chassis, this will show the status as `down`. The status will also be `down` for fabric links that are down due to some other physical error. To identify links that are down due to error vs links that are not expected to be up because of the chassis connectivity, we need to build up a list of expected fabric connectivity for each ASIC. This can be computed ahead of time based on the vendor configuration and populated in the minigraph. This will be implemented in a later phase.
 
+## 2.8 Fabric Link Monitor
+
+SONiC needs to monitor the fabric link status and take corresponding actions once an unhealthy link is detected to avoid traffic loss. Once the fabric link monitoring feature is enabled, SONiC needs to monitor the fabric capacity of a forwarding ASIC and take corresponding action once the capacity goes below the configured threshold.
+
+The design of fabric link monitor is intentionally scoped to use local component state such as information local to a linecard or information local to a supervisor. This design simplifies the need for inter-component communication.
+
+### 2.8.1 Monitor Fabric Link Status
+
+Unhealthy fabric links may lead to traffic drops. Fabric link monitoring is an important tool to minimize traffic loss. The fabric link monitor algorithm monitors fabric link status and takes links out of service if one or more criteria are true. This is needed on both fabric ASICs and forwarding ASICs.
+
+#### 2.8.1.1 Cli commands
+
+```
+> config fabric port set [port_id] isolate
+```
+
+```
+> config fabric port remove [port_id] isolate
+```
+
+The above two commands are added and can be used to manually isolate and unisolate a fabric link ( i.e. take the link out of service and put the link back into service ).
+
+#### 2.8.1.2 Fabric link monitoring criteria
+
+The fabric link monitoring algorithm checks two type of errors on a link: crc errors and uncorrectable errors. 
+
+The criteria can be extended to include checking other errors later.
+
+#### 2.8.1.2 Monitoring algorithm
+
+Instead of reacting to the counter changes, Orchagent adds a new poller and periodically polls status of all fabric links. By default, the total number of received cells, cells with crc errors, cells with uncorrectable errors are fetched from all serdes links periodically and the error rates are caculated using these numbers. If any one of the error rate is above the threshold for a number of consecutive polls, the link is identified as a unhealthy link. Then the link is automatically isolated to not distribute traffic.
```

Comment:
In the above section CLI command is defined to isolate a fabric link, in this section it is mentioned `The link is automatically isolated to not distribute traffic. ` If the links isolated automatically, can elaborate  the use case of the isolate CLIs ?

Reply:
Great question.  The fabric inks are isolated automatically by the algorithm if they are unhealthy.  

The two commands are kind of additional debug tool.  The commands can be used to manually isolate and unisolate a fabric link and can help us debug on the system as well as force isolate a fabric link.

Reply:
@kenneth-arista , @jfeng-arista 

According to this document, currently if fabric links are bad the links are isolated automatically by SDK. Can you confirm this?

Reply:
If the monitoring algorithm detected the link, and considered it as unhealthy, the algorithm calls sdk api to isolate the link, users do not need issue any CLI.